index.js

```diff
@@ -42,3 +48,7 @@ module.exports = function (func, cb) {
     return deferred.promise;
   }
 };
+
+module.exports.cb = function (func, cb) {
+  return runAsync(func, cb, true);
```

Comment:
So this is what @SBoudrias also meant.
This implementation is much better. Thank you!